File: base/src/main/java/com/explorer/base/recyclerview/BaseViewHolder.java
package com.explorer.base.recyclerview;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.explorer.base.customview.BaseCustomViewModel;
import com.explorer.base.customview.ICustomView;

public class BaseViewHolder extends RecyclerView.ViewHolder {

    ICustomView view;

    public BaseViewHolder(ICustomView view) {
        super((View) view);
        this.view = view;
    }

    public void bind(@NonNull BaseCustomViewModel item) {
        view.setData(item);
    }
}

File: app/src/main/java/com/explorer/wallettest/ui/base/BaseViewHolder.kt
package com.explorer.wallettest.ui.base

import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * @author weixuefeng@diynova.com
 * @version $
 * @time: 3/24/21--6:17 PM
 * @description
 * @copyright (c) 2021 Newton Foundation. All rights reserved.
 */
open class BaseViewHolder<T>(private val iCustomView: ICustomView<T>)
    : RecyclerView.ViewHolder(iCustomView as View) {

    fun bind(data: T) {
        iCustomView.setData(data)
    }
}
